### ISSUE TITLE
[java] InvalidLogMessageFormat fails to detect multi-stage string concatenations

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#4862]
**Affects PMD Version:** 
7.0.0

**Rule:**

[InvalidLogMessageFormat](https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#invalidlogmessageformat)

**Description:**

**Code Sample demonstrating the issue:**

```java
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Slf4JMessageFormatConcatenation {
    private static final Logger LOG = LoggerFactory.getLogger(Slf4JMessageFormatConcatenation.class);

    public void run() {
        String message = "Field: {}";
        // some code
        String error = message + ", no error";

        // log message: "Field: {}, no error"
        LOG.error(error, "fieldValue", "errorMessage"); // 1 placeholder, 2 args
    }
}
```

**Expected outcome:**

1 violation at line 13

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

CLI
